systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java

```diff
@@ -529,39 +532,38 @@ protected void testDockerImagesForKafkaCluster(String clusterName, String cluste
-            if (StUtils.isParallelNamespaceTest(extensionContext)) {
-                // if RBAC is enable we don't run tests in parallel mode and with that said we don't create another namespaces
-                if (!Environment.isNamespaceRbacScope()) {
-                    final String namespaceToDelete = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.NAMESPACE_KEY).toString();
-
-                    LOGGER.info("Deleting namespace:{} for test case:{}", namespaceToDelete, extensionContext.getDisplayName());
-                    cluster.deleteNamespace(CollectorElement.createCollectorElement(extensionContext.getRequiredTestClass().getName(), extensionContext.getDisplayName()), namespaceToDelete);
-                }
-            }
+            testSuiteNamespaceManager.deleteParallelNamespace(extensionContext);
         }
     }
 
-    protected synchronized void afterAllMayOverride(ExtensionContext extensionContext) throws Exception {
-        if (!Environment.SKIP_TEARDOWN) {
-            ResourceManager.getInstance().deleteResources(extensionContext);
-        }
+    private final void afterAllMustExecute(ExtensionContext extensionContext) {
         if (StUtils.isParallelSuite(extensionContext)) {
-            ParallelSuiteController.removeParallelSuite(extensionContext);
+            parallelSuiteController.removeParallelSuite(extensionContext);
         }
 
+        if (StUtils.isIsolatedSuite(extensionContext)) {
+            parallelSuiteController.unLockIsolatedSuite();
+        }
         // 1st case = contract that we always change configuration of CO when we annotate suite to 'isolated' and therefore
         // we need to rollback to default configuration, which most of the suites use.
         // ----
         // 2nd case = transition from if previous suite is @IsolatedSuite and now @ParallelSuite is running we must do
         // additional check that configuration is in default
-        if (install != null && !SetupClusterOperator.defaultInstallation().createInstallation().equals(install)) {
+        if (clusterOperator != null && !clusterOperator.defaultInstallation().createInstallation().equals(clusterOperator)) {
             // install configuration differs from default one we are gonna roll-back
             LOGGER.info(String.join("", Collections.nCopies(76, "=")));
-            LOGGER.info("Configurations of previous Cluster Operator are not identical. Starting rollback to the default configuration.");
+            LOGGER.info("{} - Configurations of previous Cluster Operator are not identical. Starting rollback to the default configuration.", extensionContext.getRequiredTestClass().getSimpleName());
+            LOGGER.info("Current Cluster Operator configuration:\n" + clusterOperator.toString());
+            LOGGER.info("Default Cluster Operator configuration:\n" + clusterOperator.defaultInstallation().createInstallation().toString());
```

Comment:
Could you please show me the example of output? Wouldn't be better to show just some diff?

Reply:
```java
2021-12-18T19:55:57.4718834Z 2021-12-18 19:55:57 [main] INFO  [AbstractST:556] Current Cluster Operator configuration:
2021-12-18T19:55:57.4721507Z SetupClusterOperator{cluster=io.strimzi.test.k8s.KubeClusterResource@4829e0f5, extensionContext=org.junit.jupiter.engine.descriptor.JupiterEngineExtensionContext@de1451e, clusterOperatorName='strimzi-cluster-operator', namespaceInstallTo='infra-namespace', namespaceToWatch='*', bindingsNamespaces=[infra-namespace], operationTimeout=30000, reconciliationInterval=30000, extraEnvVars=[], extraLabels={}, clusterOperatorRBACType=CLUSTER, testClassName='null', testMethodName='null'}
2021-12-18T19:55:57.4723473Z 2021-12-18 19:55:57 [main] INFO  [AbstractST:557] Default Cluster Operator configuration:
2021-12-18T19:55:57.4725999Z SetupClusterOperator{cluster=io.strimzi.test.k8s.KubeClusterResource@4829e0f5, extensionContext=org.junit.jupiter.engine.descriptor.JupiterEngineExtensionContext@de1451e, clusterOperatorName='strimzi-cluster-operator', namespaceInstallTo='infra-namespace', namespaceToWatch='*', bindingsNamespaces=[infra-namespace], operationTimeout=300000, reconciliationInterval=30000, extraEnvVars=[], extraLabels={}, clusterOperatorRBACType=CLUSTER, testClassName='null', testMethodName='null'}
```

Reply:
I can only print the diff as you suggested and it could look like:
```java
2021-12-18T19:55:57.4723473Z 2021-12-18 19:55:57 [main] INFO  [AbstractST:557] Current Cluster Operator configuration differs from default Cluster Operator in these attributes:
2021-12-18T19:55:57.4725997Z 2021-12-18 19:55:59 [main] SetupClusterOperator{operationTimeout=30000}
2021-12-18T19:55:57.4725999Z 2021-12-18 19:56:00 [main] Default SetupClusterOperator{operationTimeout=300000}
```

Reply:
WDYT?

Reply:
What about whole config on DEBUG level and just DIFF on INFO? With some proper log message that this is a diff

Reply:
Ok :) I have implement it ;) 
```java
2022-01-04 23:56:48 [main] DEBUG [SetupClusterOperator:676] ============================================================================
2022-01-04 23:56:48 [main] DEBUG [SetupClusterOperator:677] eeee - Configurations of previous Cluster Operator are not identical. Starting rollback to the default configuration.
2022-01-04 23:56:48 [main] DEBUG [SetupClusterOperator:678] Current Cluster Operator configuration:
SetupClusterOperator{cluster=io.strimzi.test.k8s.KubeClusterResource@5149f008, extensionContext=null, clusterOperatorName='strimzi-cluster-operator', namespaceInstallTo='hello', namespaceToWatch='*', bindingsNamespaces=[hello], operationTimeout=3000, reconciliationInterval=10, extraEnvVars=[], extraLabels={}, clusterOperatorRBACType=CLUSTER, testClassName='null', testMethodName='null'}
2022-01-04 23:56:48 [main] DEBUG [SetupClusterOperator:679] Default Cluster Operator configuration:
SetupClusterOperator{cluster=io.strimzi.test.k8s.KubeClusterResource@5149f008, extensionContext=null, clusterOperatorName='strimzi-cluster-operator', namespaceInstallTo='infra-namespace', namespaceToWatch='*', bindingsNamespaces=[infra-namespace], operationTimeout=300000, reconciliationInterval=30000, extraEnvVars=[], extraLabels={}, clusterOperatorRBACType=CLUSTER, testClassName='null', testMethodName='null'}
2022-01-04 23:56:48 [main] INFO  [SetupClusterOperator:680] Current Cluster Operator configuration differs from default Cluster Operator in these attributes:
operationTimeout=3000, reconciliationInterval=10, namespaceInstallTo=hello, bindingsNamespaces=[hello], 
2022-01-04 23:56:48 [main] DEBUG [SetupClusterOperator:681] ============================================================================
```